kessel-sdk/src/main/java/org/project_kessel/api/rbac/v2/ListWorkspaces.java

```diff
@@ -0,0 +1,125 @@
+
+    private static final int DEFAULT_PAGE_LIMIT = 1000;
+
+    public static Iterable<StreamedListObjectsResponse> listWorkspaces(
+            KesselInventoryServiceBlockingStub inventory,
+            SubjectReference subject,
+            String relation) {
+        return listWorkspaces(inventory, subject, relation, null);
+    }
+
+    public static Iterable<StreamedListObjectsResponse> listWorkspaces(
+            KesselInventoryServiceBlockingStub inventory,
+            SubjectReference subject,
+            String relation,
+            String continuationToken) {
+        return () -> new WorkspaceListIterator(inventory, subject, relation, continuationToken);
+    }
+
+    private static class WorkspaceListIterator implements Iterator<StreamedListObjectsResponse> {
+        private KesselInventoryServiceBlockingStub inventory;
+        private SubjectReference subject;
+        private String relation;
+
+        private String continuationToken;
+        private String lastToken;
+        private Iterator<StreamedListObjectsResponse> currentPageIterator;
+        private StreamedListObjectsResponse nextResponse;
+
+        public WorkspaceListIterator(
+                KesselInventoryServiceBlockingStub inventory,
+                SubjectReference subject,
+                String relation,
+                String initialToken) {
+            this.inventory = inventory;
+            this.subject = subject;
+            this.relation = relation;
+            this.continuationToken = initialToken;
+            this.lastToken = null;
+            this.currentPageIterator = Collections.emptyIterator();
+        }
+
+        @Override
+        public boolean hasNext() {
+            if (nextResponse != null) {
+                return true;
+            }
+            return tryToFetchNext();
+        }
+
+        @Override
+        public StreamedListObjectsResponse next() {
+            if (nextResponse == null && !tryToFetchNext()) {
+                throw new NoSuchElementException("No more workspace responses available");
+            }
+            StreamedListObjectsResponse response = nextResponse;
+            nextResponse = null;
+            return response;
+        }
+
+        private boolean tryToFetchNext() {
+            while (true) {
+                if (currentPageIterator.hasNext()) {
+                    nextResponse = currentPageIterator.next();
+                    if (nextResponse.hasPagination()) {
+                        String token = nextResponse.getPagination().getContinuationToken();
+                        if (token != null) {
+                            lastToken = token;
+                        }
+                    }
+                    return true;
+                }
+
+                if (lastToken != null && lastToken.isEmpty()) {
+                    return false;
+                }
+
+                if (lastToken != null) {
+                    continuationToken = lastToken;
```

Comment:
Why do we have `continuationToken` and `lastToken` ? 

I think we could simplify some of this logic if we have `continuationToken = nextResponse.getPagination().getContinuationToken() `  now that we know the token is not going to be null.